### PR TITLE
Add com.neatdecisions.Detwinner

### PR DESCRIPTION
**Pull request by neatdecisions**
Hello,

Detwinner is a GTK3-based tool to search and remove exact file duplicates and similar images.
Upstream repository: [https://github.com/neatdecisions/detwinner](https://github.com/neatdecisions/detwinner)
Depends on:
- gtkmm (et al.)
- GraphicsMagick

### CODE REVIEW

**Comment by LeandroStanger** [comment]
bot, build com.neatdecisions.Detwinner

File: com.neatdecisions.Detwinner.json [hunk 1]
```diff
@@ -0,0 +1,180 @@
+{
+  "app-id": "com.neatdecisions.Detwinner",
+  "runtime": "org.gnome.Sdk",
+  "runtime-version": "3.32",
+  "sdk": "org.gnome.Sdk",
+  "command": "detwinner",
+  "finish-args": [
+    "--socket=x11",
```
Comment (nedrichards) [inline com.neatdecisions.Detwinner.json]: if you use this you also need --share=ipc

File: com.neatdecisions.Detwinner.json [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+    "--socket=wayland",
+    "--filesystem=host",
+    "--filesystem=xdg-config/gtk-3.0",
+    "--talk-name=org.gtk.vfs",
+    "--talk-name=org.gtk.vfs.*",
+    "--talk-name=org.freedesktop.portal.Trash"
+  ],
+  "build-options": {
```
Comment (nedrichards) [inline com.neatdecisions.Detwinner.json]: these options are included by default in this version of the SDK and it's redundant here.

File: com.neatdecisions.Detwinner.json [hunk 3]
```diff
@@ -0,0 +1,180 @@
...
+      ]
+    },
+    {
+      "name": "detwinner",
+      "buildsystem": "meson",
+      "builddir": true,
+      "config-opts": [
+        "--buildtype=release"
+      ],
+      "build-options": {
```
Comment (nedrichards) [inline com.neatdecisions.Detwinner.json]: duplicated from above, and, as mentioned there redundant.

**Comment by neatdecisions** [comment]
Thanks for the review! Added --share=ipc, removed build-options (and config-opts for the main source).

**Comment by LeandroStanger** [comment]
bot, build com.neatdecisions.Detwinner

File: com.neatdecisions.Detwinner.json [hunk 4]
```diff
@@ -0,0 +1,167 @@
...
+  "runtime": "org.gnome.Sdk",
+  "runtime-version": "3.32",
+  "sdk": "org.gnome.Sdk",
+  "command": "detwinner",
+  "finish-args": [
+    "--share=ipc",
+    "--socket=x11",
+    "--socket=wayland",
+    "--filesystem=host",
+    "--filesystem=xdg-config/gtk-3.0",
```
Comment (bilelmoussaoui) [inline com.neatdecisions.Detwinner.json]: Why it does need this one?

Comment (AsciiWolf) [reply com.neatdecisions.Detwinner.json]: Maybe use `--persist` instead?

Comment (AsciiWolf) [reply com.neatdecisions.Detwinner.json]: But I am not sure if --persist can be used in combination with host file system access.

Comment (neatdecisions) [reply com.neatdecisions.Detwinner.json]: In fact without this permission the user's bookmarks won't be shown in `GtkFileChooser` and `GtkPlacesSidebar`, even with `--filesystem=host`. Is there any better way to fix it?

**Comment by nedrichards** [comment]
Repository has been created: https://github.com/flathub/com.neatdecisions.Detwinner

**Pull request closed without merging**